vmss-prototype/vmss-prototype

```diff
@@ -829,25 +829,26 @@ def vmss_prototype_update(sub_args):
                 '--name', extension['name']
                 ], retries=3, check=False, retry_func=not_found_no_retry)
 
-    # Scale out 1 more VMSS instance to replace the previously deleted instance
-    # Note: this does not respect any changes that may have occured to the VMSS instance count during update
-    # from out-of-band node scaling tools such as cluster-autoscaler
-    output, _, _ = run(az(['vmss', 'show'], subscription) + [
-                       '--resource-group', resource_group,
-                       '--name', vmss
-                       ], retries=3)
-    vmss_info = json.loads(output)
-    capacity = vmss_info["sku"]["capacity"]
-    if isinstance(capacity, int):
-        # Increase the capacity of the VMSS by 1
-        run(az(['vmss', 'update'], subscription) + [
-            '--resource-group', resource_group,
-            '--name', vmss,
-            '--set',
-                'sku.capacity={}'.format(capacity + 1),
-            ], retries=3, check=False)
-    else:
-        logging.warning('Unable to determine capacity for VMSS {}, will not add back 1 instance'.format(vmss)
+    if sub_args.new_updated_nodes > 0:
+        # Scale out 1 more VMSS instance to replace the previously deleted instance
+        # Note: this does not respect any changes that may have occured to the VMSS instance count during update
+        # from out-of-band node scaling tools such as cluster-autoscaler
+        output, _, _ = run(az(['vmss', 'show'], subscription) + [
+                        '--resource-group', resource_group,
+                        '--name', vmss
+                        ], retries=3)
+        vmss_info = json.loads(output)
+        capacity = lookup_one_value(vmss_info,'sku.capacity')
+        if capacity != None:
+            # Increase the capacity of the VMSS by the value of --new-updated-nodes
+            run(az(['vmss', 'update'], subscription) + [
+                '--resource-group', resource_group,
+                '--name', vmss,
+                '--set',
```

Comment:
Do we want to add --no-wait to this such that we don't wait until the capacity change completes?

This can take multiple minutes to complete as it can wait for the VM to actually start (finish provisioning).

Reply:
Yes, good catch